### PR TITLE
File operations

### PR DESCRIPTION
**Pull request by masroorhasan**
Working implementation of fsOpen() and fsClose()

### CODE REVIEW

File: client/client.c [hunk 1]
```diff
@@ -22,24 +22,58 @@ int main(int argc, char *argv[]) {
     }
 
     fsMount(argv[1], atoi(argv[2]), "leaf");
+    // fsMount(argv[1], atoi(argv[2]), "leaf");
```
Comment (naturalwarren) [inline client/client.c]: Stray comment.

File: client/client.c [hunk 2]
```diff
@@ -22,24 +22,58 @@ int main(int argc, char *argv[]) {
...
-    fsMount(argv[1], atoi(argv[2]), "leaf");
+    char* folder_path = "leaf/one";
+    FSDIR* fdir = fsOpenDir(folder_path);
 
-    fsUnmount("xyz");
+    struct fsDirent *fdent = NULL;
 
-    if(fd != NULL) {
-        struct fsDirent *fdent = NULL;
+    // for(fdent = fsReadDir(fdir); fdent != NULL; fdent = fsReadDir(fdir)) {
```
Comment (naturalwarren) [inline client/client.c]: Stray comments.

File: client/client.c [hunk 3]
```diff
@@ -22,24 +22,58 @@ int main(int argc, char *argv[]) {
...
+    fdent = fsReadDir(fdir);
+    printf("%s, %d\n", fdent->entName, (int)(fdent->entType));
 
-        int ret = fsCloseDir(fd);
-        fdent = fsReadDir(fd);
-    }
+    fdent = fsReadDir(fdir);
+    printf("%s, %d\n", fdent->entName, (int)(fdent->entType));
+
+    // open bunch of files
```
Comment (naturalwarren) [inline client/client.c]: Single line comments start with capitals.

File: client/ece454_fs_client.c [hunk 4]
```diff
@@ -247,7 +247,35 @@ extern struct fsDirent *fsReadDir(FSDIR * folder) {
 extern int fsOpen(const char *fname, int mode) {
     // Check that we're mounted
     if (mountError(true)) return -1;
-    return -1;
+    printf("fname: %s, mode %i\n", fname, mode);
```
Comment (naturalwarren) [inline client/ece454_fs_client.c]: Remove this print and put it after the RPC call like all the others.

File: server/ece454_fs_server_util.c [hunk 5]
```diff
@@ -75,3 +75,9 @@ extern return_type serializeFsDirent(struct dirent *d) {
     return ret;
 }
 
+/*
```
Comment (naturalwarren) [inline server/ece454_fs_server_util.c]: Stray comment with no method definition. Remove.

File: server/ece454_fs_server.c [hunk 6]
```diff
@@ -236,8 +236,65 @@ extern return_type fsReadDir(const int nparams, arg_type *a) {
...
-    return r;
+    printf("fsOpen() called.\n");
+
+    int fname_sz = a->arg_size;
+
+    // Get filename to open
+    char *fname = (char *) malloc(fname_sz);
+    memcpy(fname, (char *)a->arg_val, fname_sz);
+
+    // Parse file name with server path
```
Comment (naturalwarren) [inline server/ece454_fs_server.c:254]: Can we comment block this the same way we did in fsOpen?

File: client/client.c (hunk 1 above)
Comment (masroorhasan) [inline client/client.c]: Updated.

File: client/client.c (hunk 2 above)
Comment (masroorhasan) [inline client/client.c]: Updated.

File: client/client.c (hunk 3 above)
Comment (masroorhasan) [inline client/client.c]: Updated.

File: client/ece454_fs_client.c (hunk 4 above)
Comment (masroorhasan) [inline client/ece454_fs_client.c]: Deleted.

File: server/ece454_fs_server_util.c (hunk 5 above)
Comment (masroorhasan) [inline server/ece454_fs_server_util.c]: Updated.

File: server/ece454_fs_server.c (hunk 6 above)
Comment (masroorhasan) [inline server/ece454_fs_server.c:254]: Updated.

**Pull request merged**